codalab/client/local_bundle_client.py

```diff
@@ -167,12 +167,16 @@ def get_bundle_target(self, target):
         return (self.model.get_bundle(bundle_uuid), subpath)
 
     def get_worksheet_uuid(self, base_worksheet_uuid, worksheet_spec):
-        if worksheet_spec == '' or worksheet_spec == worksheet_util.HOME_WORKSHEET:
```

Comment:
You still want to intercept '' and HOME_WORKSHEET and handle them specially as before - these should not get passed into canonicalize.get_worksheet_uuid.

Then the dashboard creation can be in the UsageError as you have it.